nemo/collections/nlp/models/enc_dec_nlp_model.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import MISSING, dataclass
+from typing import Any
+
+from pytorch_lightning.trainer.trainer import Trainer
+
+from nemo.collections.nlp.models.nlp_model import NLPModel
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_tokenizer
+
+
+@dataclass
+class EncDecNLPModelConfig:
+    encoder_tokenizer: Any = MISSING
+    decoder_tokenizer: Any = MISSING
+    encoder_embedding: Any = MISSING
+    decoder_embedding: Any = MISSING
+    encoder: Any = MISSING
+    decoder: Any = MISSING
+    head: Any = MISSING
+
+
+class EncDecNLPModel(NLPModel):
+    """Base class for encoder-decoder NLP models.
+    """
+
+    def __init__(self, cfg: EncDecNLPModelConfig, trainer: Trainer = None):
+        super().__init__(cfg=cfg, trainer=trainer)
+
+    @property
```

Comment:
I think having setter for encoder/decoder_vocab_size is error prone.
The user may do ``model.decoder_vocab_size=20000`` but nothing else and expect decoder's tokenizer to actually change to 20K vocab. But it won't and won't throw an error.

How about we remove enconder/decoder_vocab_size and their setters?

Reply:
Or a better solution to actually have only getters (not setters) and return there 
``
self.encoder/decoder_tokenizer.vocab_size
``